### PR TITLE
Update react-instantsearch-core types

### PR DESCRIPTION
**Pull request by davidfurlong**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://community.algolia.com/react-instantsearch/connectors/connectInfiniteHits.html https://github.com/algolia/react-instantsearch/blob/c2df206919b0fd8bcf5aa1e3f265f56fc2476ef9/packages/react-instantsearch-core/src/connectors/connectRefinementList.js#L152
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by davidfurlong** [comment]
@gburgett @jpowell

File: types/react-instantsearch-core/index.d.ts [hunk 1]
```diff
@@ -171,13 +172,21 @@ export function connectHighlight(Composed: React.ComponentType<any>): React.Comp
...
-export function connectHits<TProps extends { hits: THit[]}, THit>(ctor: React.ComponentType<TProps>): ConnectedComponentClass<TProps, {hits?: THit[]}>;
+export function connectHits<TProps extends { hits: THit[] }, THit>(ctor: React.ComponentType<TProps>): ConnectedComponentClass<TProps, { hits?: THit[] }>;
 
 export function connectHitsPerPage(Composed: React.ComponentType<any>): React.ComponentClass<any>;
-export function connectInfiniteHits(Composed: React.ComponentType<any>): React.ComponentClass<any>;
+
+export interface InfiniteHitsProvided<THit = any> {
+  hits: THit[];
+  refine: (...args: any[]) => any;
+  hasMore: boolean;
```
Comment (gburgett) [inline types/react-instantsearch-core/index.d.ts]: Thanks for the PR!  Can you please add the documentation for these properties from their API documentation? https://community.algolia.com/react-instantsearch/connectors/connectInfiniteHits.html

File: types/react-instantsearch-core/index.d.ts [hunk 2]
```diff
@@ -256,6 +265,8 @@ export interface RefinementListProvided {
   searchForItems: (...args: any[]) => any;
   /** a boolean that says if the items props contains facet values from the global search or from the search inside items. */
   isFromSearch: boolean;
+  /** a boolean that says whether you can currently refine */
+  canRefine: boolean;
```
Comment (gburgett) [inline types/react-instantsearch-core/index.d.ts]: Thanks for catching that!

**Review by gburgett — CHANGES_REQUESTED** [review]
Looks like if you can copy paste that doc then we'll be good to go.

**Comment by davidfurlong** [comment]
> Looks like if you can copy paste that doc then we'll be good to go.

Fixed @gburgett

**Review by gburgett — APPROVED** [review]
Thanks @davidfurlong !

**Comment by PranavSenthilnathan** [comment]
Congratulations on your first DefinitelyTyped contribution! Thank you for being a part of the community!

**Pull request merged**